rrd/rrd_interface.ml

```diff
@@ -32,92 +35,375 @@ let set_sockets_dir x =
+        (uid_p @-> returning float_p rrd_err)
+    end
+
+    module Interdomain = struct
+
+      let iduid_p = Param.mk ~name:"uid" ~description:[ "Interdomain ID" ] interdomain_uid
+
+      let register =
+        declare "Plugin.Interdomain.register"
+          [ "[Plugin.Interdomain.register uid info protocol] registers an"
+          ; "interdomain plugin. [uid] is the unique identifier of the"
+          ; "plugin, containing both a name and the frontend domain id."
+          ; "[info] contains both the desired sampling frequency and"
+          ; "a list of the grant references of the shared pages. [protocol]"
+          ; "is V1 or V2, and the return value is the time until the next"
+          ; "reading" ]
+          (iduid_p @-> info_p @-> protocol_p @-> returning float_p rrd_err)
+
+      let deregister = declare "Plugin.Interdomain.deregister"
+        [ "Deregisters a plaugin by uid." ]
+        (iduid_p @-> returning unit_p rrd_err)
+
+      let next_reading = declare "Plugin.Interdomain.next_reading"
+        [ "Returns the number of seconds before the next reading." ]
+        (iduid_p @-> returning float_p rrd_err)
+    end
+
+    let register =
+      let freq_p = Param.mk ~name:"frequency" ~description:[ "Rrd database sampling frequency" ] rrd_freq in
+      declare "Plugin.register"
+        [ "Preserved for backwards compatibility. Equivalent to a Local"
+        ; "plugin registration with V1 protocol." ]
+        (uid_p @-> freq_p @-> returning float_p rrd_err)
+
+    let deregister = declare "Plugin.deregister"
+      [ "Preserved for backwards compatibility. Deregesters a local plugin." ]
+      (uid_p @-> returning unit_p rrd_err)
+
+    let next_reading = declare "Plugin.next_reading"
+      [ "Returns the time until the next reading." ]
+      (uid_p @-> returning float_p rrd_err)
+  end
+
+  (** High availability module *)
+  module HA = struct
+
+    let enable_and_update =
+      let heartb_lat_p  = Param.mk ~name:"heartbeat_latency"   ~description:[ "Time taken for heartbeat signal to travel" ] Types.float in
+      let xapi_lat_p    = Param.mk ~name:"xapi_latency"        ~description:[ "Time taken for Xapi to respond" ]            Types.float in
+      let stfile_lats_p = Param.mk ~name:"statefile_latencies" ~description:[ "Time taken for statefiles to update" ]       sflat_lst in
+      declare "HA.enable_and_update"
+        [ "Enables the gathering of HA metrics, a built in function of xcp-rrdd." ]
+        (stfile_lats_p @-> heartb_lat_p @-> xapi_lat_p @-> returning unit_p rrd_err)
+
+    let disable = declare "HA.disable"
+      [ "Disables the HA metrics." ]
+      (unit_p @-> returning unit_p rrd_err)
+  end
+
+  module Deprecated = struct
 
-module Deprecated = struct
-	(* Could change timescale to sum type, e.g. Slow | Fast.*)
-	external load_rrd : uuid:string -> timescale:int ->
-		master_address:string option -> unit = ""
+    let load_rrd = (* TODO: there can only be one *)
```

Comment:
What is the TODO about?

Reply:
Ah, that can be removed now. I think this was an issue with namespaces but it's been resolved now.